src/app/containers/MostWatched/utilities/index.js

```diff
@@ -0,0 +1,51 @@
+import pathOr from 'ramda/src/pathOr';
+import filterPopularStaleData from '#app/lib/utilities/filterPopularStaleData';
+import {
+  MOST_WATCHED_CLIENT_REQUEST,
+  MOST_WATCHED_FETCH_ERROR,
+} from '#lib/logger.const';
+import getMostWatchedEndpoint from '#lib/utilities/getMostWatchedUrl';
+import nodeLogger from '#lib/logger.node';
+
+const logger = nodeLogger(__filename);
+
+export const processMostWatched = ({ data, isAmp, numberOfItems, service }) => {
+  if (!data) {
+    return null;
+  }
+  const filteredData = filterPopularStaleData({
+    data,
+    isAmp,
+    service,
+    popularType: 'mostWatched',
+  });
+
+  if (!filteredData) {
+    return null;
+  }
+
+  const records = pathOr([], ['records'], filteredData);
+  return records.slice(0, numberOfItems).map(item => item.promo);
+};
+
+export const getMostWatchedData = async ({ service, variant }) => {
+  const endpoint = getMostWatchedEndpoint({ service, variant });
+  logger.info(MOST_WATCHED_CLIENT_REQUEST, { url: endpoint });
```

Comment:
It would also be useful to know the path which caused the most watched request to be invoked (i.e. MAP page path) - can this be passed in from somewhere?

Reply:
Done